packages/beacon-node/src/network/reqresp/handlers/dataColumnSidecarsByRange.ts

```diff
@@ -0,0 +1,166 @@
+  columns: ColumnIndex[]
+): Iterable<ResponseOutgoing> {
+  const retrivedColumnsLen = ssz.Uint8.deserialize(
+    dataColumnSidecarsBytesWrapped.slice(NUM_COLUMNS_IN_WRAPPER_INDEX, COLUMN_SIZE_IN_WRAPPER_INDEX)
+  );
+  const retrievedColumnsSizeBytes = dataColumnSidecarsBytesWrapped.slice(
+    COLUMN_SIZE_IN_WRAPPER_INDEX,
+    CUSTODY_COLUMNS_IN_IN_WRAPPER_INDEX
+  );
+  const columnsSize = ssz.UintNum64.deserialize(retrievedColumnsSizeBytes);
+  const dataColumnsIndex = dataColumnSidecarsBytesWrapped.slice(
+    CUSTODY_COLUMNS_IN_IN_WRAPPER_INDEX,
+    CUSTODY_COLUMNS_IN_IN_WRAPPER_INDEX + NUMBER_OF_COLUMNS
+  );
+  const allDataColumnSidecarsBytes = dataColumnSidecarsBytesWrapped.slice(
+    DATA_COLUMN_SIDECARS_IN_WRAPPER_INDEX + 4 * retrivedColumnsLen
+  );
+
+  const columnsLen = allDataColumnSidecarsBytes.length / columnsSize;
+
+  // const storedColumns = Array.from({length: NUMBER_OF_COLUMNS}, (_v, i) => i).filter((i) => dataColumnsIndex[i] > 0);
+  // console.log(
+  //   `onDataColumnSidecarsByRange: slot=${blockSlot} columnsSize=${columnsSize} columnsLen=${columnsLen} retrivedColumnsLen=${retrivedColumnsLen} toredColumnsNum=${allDataColumnSidecarsBytes.length / columnsSize}, storedColumns=${storedColumns.join(" ")}`
+  // );
+
+  // no columns possibly no blob
+  if (columnsLen === 0) {
+    return;
+  }
+
+  const fork = chain.config.getForkName(blockSlot);
+
+  for (const index of columns) {
+    // get the index at which the column is
+    const dataIndex = (dataColumnsIndex[index] ?? 0) - 1;
+    if (dataIndex < 0) {
+      throw new ResponseError(
+        RespStatus.SERVER_ERROR,
+        `dataColumnSidecar index=${index} dataIndex=${dataIndex} not custodied`
+      );
+    }
+    const dataColumnSidecarBytes = allDataColumnSidecarsBytes.slice(
+      dataIndex * columnsSize,
+      (dataIndex + 1) * columnsSize
+    );
+    if (dataColumnSidecarBytes.length !== columnsSize) {
+      throw new ResponseError(
+        RespStatus.SERVER_ERROR,
+        `Invalid dataColumnSidecar index=${index} dataIndex=${dataIndex} bytes length=${dataColumnSidecarBytes.length} expected=${columnsSize} for slot ${blockSlot} blobsLen=${columnsLen}`
+      );
+    }
+    // console.log(`iterate onDataColumnSidecarsByRange blockSlot=${blockSlot} index=${index} dataIndex=${dataIndex}`);
+    yield {
+      data: dataColumnSidecarBytes,
+      fork,
+    };
+  }
+}
+
+export function validateDataColumnSidecarsByRangeRequest(
+  request: fulu.DataColumnSidecarsByRangeRequest
+): fulu.DataColumnSidecarsByRangeRequest {
+  const {startSlot, columns} = request;
+  let {count} = request;
+
+  if (count < 1) {
+    throw new ResponseError(RespStatus.INVALID_REQUEST, "count < 1");
+  }
+  // TODO: validate against MIN_EPOCHS_FOR_BLOCK_REQUESTS
+  if (startSlot < GENESIS_SLOT) {
+    throw new ResponseError(RespStatus.INVALID_REQUEST, "startSlot < genesis");
+  }
+
+  if (count > MAX_REQUEST_BLOCKS_DENEB) {
+    count = MAX_REQUEST_BLOCKS_DENEB;
+  }
+
```

Comment:
As this input is coming from external actor suggest to also validate the `columns`. 

```ts
if (!columns || columns.length === 0) {
  throw new ResponseError(RespStatus.INVALID_REQUEST, "columns array is empty");
}
```